### PR TITLE
Refactor advantage

### PR DESCRIPTION
**Pull request by albertkol**
## Done
There are four stages of refactoring at this point and some bug fixes:

### Stage 1:
Move everything advantage related inside an advantage folder.
Fixing small bugs I found:
1) `get_cutomer_info` throws 500 if `user_token` is missing
2) `post_advantage_subscription` for already existing product did not increment the existing value correctly

### Stage 2: 
Add decorator `advantage_checks`:
1) Set the env: staging or prod `api_url`
2) Set common advantage views kwargs: `api_url`, `token`, `token_type`, `is_test_backend`
3) Set parameter validation for all views using `webargs`

### Stage 3: 
Incorporate `store_maintenance` in `advantage_checks_decorator`.

### Stage 4:
Make `get_account_subscriptions_for_marketplace` to return the dict
inside the `subscriptions`.
Fix resizing bug. Before it would add up the number selected to what the
user already had. Now it's correctly overriding it.

### Stage 5
    
Move the try and except logic from the views to the api class.
    
Create 4 differnt kind of Exceptions that we catch on an app level:
    1) `UAContractsAPIError` - throws json error after the sentry capture
    2) `UAContractsAPIErrorView` - returns the 500 html template after
       sentry capture
    3) `UAContractsAPIAuthError` - clears `flask.session` and reutrns json
       after sentry capture
    4) `UAContractsAPIAuthErrorView` - clears `flask.session` and refreshes
       page after sentry capture

### CODE REVIEW

File: webapp/decorators.py [hunk 1]
```diff
@@ -24,16 +24,67 @@ def is_user_logged_in(*args, **kwargs):
...
+            )
+
+            api_url = flask.current_app.config["CONTRACTS_LIVE_API_URL"]
+
+            if is_test_backend:
+                stripe_publishable_key = os.getenv(
+                    "STRIPE_TEST_PUBLISHABLE_KEY",
+                    "pk_test_yndN9H0GcJffPe0W58Nm64cM00riYG4N46",
+                )
+                api_url = flask.current_app.config["CONTRACTS_TEST_API_URL"]
```
Comment (carkod) [inline webapp/decorators.py:52]: I suppose this test backend is using test a fake key, so people can't just get through stripe and get stuff for free

**Review by carkod — APPROVED** [review]
LGTM. If @nottrobin has anything to add?

**Comment by nottrobin** [comment]
I'm sorry I haven't got to this yet. I would like to take a look before it's merged if you don't mind...

**Comment by albertkol** [comment]
@nottrobin this is not getting merged to master. I'd like this merged asap because it's blocking any further back-end dev. Also includes some bug fixes for the front-end.

We can always tackle new code improvements for advantage in new PRs, since this feature branch is not getting merged to master any time soon.

**Comment by nottrobin** [comment]
Ok, merge it

**Pull request merged**